File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/iterators/LinearPhotoIterator.java
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.iterators;

/**
 * Created on 17/10/2018.
 */
public class LinearPhotoIterator extends AbstractPhotoIterator {

    private int _currentIndex = 0;

    @Override
    protected int getNextIndex() {
        _currentIndex = (_currentIndex + 1) % getTotalCount();

        return _currentIndex;
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/iterators/LinearPhotoIterator.kt
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.iterators

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.filters.FilterOption

/**
 * Created on 25/10/2018.
 */
class LinearPhotoIterator constructor(filterOption: FilterOption) : AbstractPhotoIterator(filterOption) {

    private var currentIndex = 0

    override val nextIndex: Int
        get() {
            currentIndex = (currentIndex + 1) % totalCount

            return currentIndex
        }
}